pipelines/common/src/main/java/com/google/fhir/analytics/JdbcConnectionPools.java

```diff
@@ -97,13 +91,15 @@ private static DataSource createNewPool(
     comboPooledDataSource.setJdbcUrl(config.jdbcUrl());
     comboPooledDataSource.setUser(config.dbUser());
     comboPooledDataSource.setPassword(config.dbPassword());
-    comboPooledDataSource.setMaxPoolSize(jdbcMaxPoolSize);
-    comboPooledDataSource.setInitialPoolSize(initialPoolSize);
-    // Setting an idle time to reduce the number of connections when idle.
-    comboPooledDataSource.setMaxIdleTime(30);
-    // Lowering the minimum pool size to limit the number of connections if multiple pools are
-    // created for the same DB.
-    comboPooledDataSource.setMinPoolSize(1);
+    comboPooledDataSource.setMaxPoolSize(Math.max(MIN_CONNECTIONS, jdbcMaxPoolSize));
+    comboPooledDataSource.setMinPoolSize(MIN_CONNECTIONS);
+    // Setting an idle time to reduce the number of connections when idle; avoid setting
+    // maxIdleTime! Instead, use connection testing as done below; see:
+    // https://www.mchange.com/projects/c3p0/#managing_pool_size
+    comboPooledDataSource.setMaxIdleTimeExcessConnections(30);
```

Comment:
Hmm, that was not my intention; I mean I want the number of connections to actually go to the minimum (e.g., 3) when there is no load. So we only maintain more connections if we really need more. I think the previous settings would achieve that but I changed the delays slightly (increasing  `IdleConnectionTestPeriod`) to prevent any confusion.

Reply:
Ok, thanks for the clarification.